BoxContentSDK/BoxContentSDK/External/HashHelper/BOXStreamingHashHelper.m

```diff
@@ -0,0 +1,48 @@
+//
+//  BOXStreamingHashHelper.m
+//  BoxContentSDK
+//
+//  Created by Prithvi Jutur on 5/21/18.
+//  Copyright © 2018 Box. All rights reserved.
+//
+
+#import "BOXHashHelper.h"
+
+#include <CommonCrypto/CommonDigest.h>
+#include <CoreFoundation/CoreFoundation.h>
+#import "BOXStreamingHashHelper.h"
+
+@interface BOXStreamingHashHelper()
+@property (nonatomic, readwrite, strong) NSString *calcSha1;
+@property (nonatomic, assign) CC_SHA1_CTX *context;
+@end
+
+@implementation BOXStreamingHashHelper
+- (instancetype) init {
+    _context = (CC_SHA1_CTX*) malloc(sizeof(CC_SHA1_CTX));
+    return self;
+}
+
+- (void) open {
+    CC_SHA1_Init(_context);
```

Comment:
Consider calling `open` in `init` or keeping track of the open state, to avoid mis-use?